src/PureScript/CST/Traversal.purs

```diff
@@ -106,22 +106,22 @@ import Data.Const (Const(..))
 import Data.Functor.Compose (Compose(..))
 import Data.Identity (Identity(..))
 import Data.Newtype (un)
-import Data.Newtype as Newtype
 import Data.Traversable (traverse)
 import Data.Tuple (Tuple(..), curry, uncurry)
+import Prim as Prim
 import PureScript.CST.Types (AdoBlock, Binder(..), CaseOf, ClassHead, DataCtor(..), DataHead, Declaration(..), Delimited, DelimitedNonEmpty, DoBlock, DoStatement(..), Expr(..), Foreign(..), Guarded(..), GuardedExpr(..), IfThenElse, Instance(..), InstanceBinding(..), InstanceHead, Labeled(..), Lambda, LetBinding(..), LetIn, Module(..), ModuleBody(..), OneOrDelimited(..), PatternGuard(..), RecordAccessor, RecordLabeled(..), RecordUpdate(..), Row(..), Separated(..), Type(..), TypeVarBinding(..), ValueBindingFields, Where(..), Wrapped(..))
 import Type.Row (type (+))
 
-type Rewrite e f g = g e -> f (g e)
-type RewriteWithContext c e f g = c -> g e -> f (Tuple c (g e))
-type MonoidalRewrite e m g = g e -> m
-type PureRewrite e g = g e -> g e
-type PureRewriteWithContext c e g = c -> g e -> Tuple c (g e)
+type Rewrite e f (g :: Prim.Type -> Prim.Type) = g e -> f (g e)
```

Comment:
This is a bit unfortunate -- adding all the kind signatures -- and I opted to do it inline because otherwise the lines are so long. That said, I can update it if you would like:

```suggestion
type Rewrite :: Prim.Type -> (Prim.TypeType -> Prim.Type) -> (Prim.Type -> Prim.Type) -> Prim.Type
type Rewrite e f g = g e -> f (g e)
```

Or, if you want polymorphic kinds:

```suggestion
type Rewrite :: forall k. k -> (Prim.Type -> Prim.Type) -> (k -> Prim.Type) -> Prim.Type
type Rewrite e f g = g e -> f (g e)
```

I could also perhaps rename the `Prim` import to be `import Prim as P` to cut the lengths a little.

Reply:
I think using `P` is fine.

Reply:
Which of the three do you prefer? Annotate inline, standalone kind signature, or standalone and leave what can be polymorphic polymorphic?

Reply:
What you've done, but with `P` is OK by me.